crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -529,6 +508,61 @@ impl BuildSpec {
 		Ok("spec")
 	}
 
+	/// Generates plain and raw chain specification files, and returns the path to the latter.
+	fn generate_chain_spec(
+		&self,
+		binary_path: &Path,
+		spinner: &ProgressBar,
+	) -> anyhow::Result<PathBuf> {
+		let BuildSpec { output_file, chain, .. } = self;
+		spinner.start("Generating chain specification...");
+
+		// Generate plain chain spec.
+		generate_plain_chain_spec(binary_path, output_file, self.default_bootnode, chain)?;
+		// Customize spec based on input.
+		self.customize()?;
+
+		// Generate raw spec.
+		spinner.set_message("Generating raw chain specification...");
+		let spec_name = &output_file
+			.file_name()
+			.and_then(|s| s.to_str())
+			.unwrap_or(DEFAULT_SPEC_NAME)
+			.trim_end_matches(".json");
+		let raw_spec_name = format!("{spec_name}-raw.json");
+		generate_raw_chain_spec(binary_path, output_file, &raw_spec_name)
+			.map_err(anyhow::Error::from)
+	}
+
+	/// Generates chain specification files and returns the file paths for the generated genesis
```

Comment:
Comment is wrong order of items in tuple, aliases can improve this.

Reply:
Outdated based on Daan's suggested refactor https://github.com/r0gue-io/pop-cli/pull/410/commits/da5f6d2ddf57e00e083d0f318403f04c916f73c6